### PR TITLE
Added support for an iOS platform

### PR DESCRIPTION
**Pull request by dukhanov**

### CODE REVIEW

**Comment by jguix** [comment]
Awesome, let me try it and I will merge your contributions.

Any special instructions (manually modifying config files, etc.) to use the plugin in iOS?

**Comment by dukhanov** [comment]
Hi,

**Comment by dukhanov** [comment]
Instructions:
- [https://github.com/alunny/node-xcode](node-xcode) should be greater than 0.8.7
- to subscribe on button event, use:

```
flicPlugin.onButtonClick({
 success: function(data){
   //do something
 },
 error: : function(data){
   //do something
  },
})
```
- button color is temporary unavailable see: https://github.com/50ButtonsEach/fliclib-ios/issues/16

I will fix all issues in the next release

File: src/ios/Flic.m [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:pluginNotInitializedMessage];
+        [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
+        return;
+    }
+    
+	[[SCLFlicManager sharedManager] grabFlicFromFlicAppWithCallbackUrlScheme:@"winfleet-tracker"];
+	result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+	[self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
+}
+
```
Comment (jguix) [inline src/ios/Flic.m:87]: I recently merged a pull request from another user who proposed changes to the JavaScript interface, such as eliminating the waitForButtonEvent and triggerButtonEvent. Now on Android it works in push mode. I guess I can delete these methods and the declaration in the .h file. Right?

**Comment by dukhanov** [comment]
Yes, sure

**Comment by jguix** [comment]
Got this error

```
Failed to install 'cordova-plugin-flic':Error: Variable(s) missing: URL_SCHEME
    at /Users/juangui/.node/lib/node_modules/cordova/node_modules/cordova-lib/src/plugman/install.js:299:23
    at _fulfilled (/Users/juangui/.node/lib/node_modules/cordova/node_modules/q/q.js:787:54)
    at self.promiseDispatch.done (/Users/juangui/.node/lib/node_modules/cordova/node_modules/q/q.js:816:30)
    at Promise.promise.promiseDispatch (/Users/juangui/.node/lib/node_modules/cordova/node_modules/q/q.js:749:13)
    at /Users/juangui/.node/lib/node_modules/cordova/node_modules/q/q.js:557:44
    at flush (/Users/juangui/.node/lib/node_modules/cordova/node_modules/q/q.js:108:17)
    at process._tickCallback (node.js:419:13)
```

**Comment by jguix** [comment]
Hi, dukhanov. I merged because it doesn't conflict with the android code, but it is not working, I get compiling errors at plugin add time (see above). Can you please review the merged code?

**Comment by dukhanov** [comment]
Hi, jguix, I forgot to mention:
For iOS platform, in your project plist file, you need to register your app to handle URL scheme calls. 
see: [https://flic.io/partners/developers/ios-tutorial]() 5th section.
To add this URL scheme automatically, you should add plugin with a variable:
`cordova plugin add https://github.com/jguix/Cordova-Flic.git --variable URL_SCHEME=youscheme`

We should update an instruction of the plugin.

**Comment by jguix** [comment]
Thanks, then please could you clarify what should be "my scheme"?

Something like?
cordova plugin add https://github.com/jguix/Cordova-Flic.git --variable
URL_SCHEME=flic

2016-05-30 9:02 GMT+02:00 Maxim notifications@github.com:

**Comment by jguix** [comment]
Or?
cordova plugin add https://github.com/jguix/Cordova-Flic.git --variable
URL_SCHEME=flic20

2016-05-30 9:47 GMT+02:00 Juangui Jordán Aldasoro juangui@gmail.com:

**Comment by dukhanov** [comment]
Yes, something like that.

Documentation from tutorial of Flic integration:

> In your project plist file, you need to register your app to handle URL scheme calls. This is needed when you fetch a Flic object from the Flic App. You also need to add the Flic app to the list of Application Query Schemes. Don't worry, we will explain this in greater detail later on. Make sure that your plist file contains something like this:
> image

**Comment by dukhanov** [comment]
I found an issue in my code, regarding URL_SCHEME, I will create PR asap.

**Comment by jguix** [comment]
Ok, thanks :)

Just a final doubt

Can I use anything as the URL_SCHEME or should I use "flic20" (I found this
string in your source code). I would like to include as much information as
possible in the README

2016-05-30 9:55 GMT+02:00 Maxim notifications@github.com:

**Comment by dukhanov** [comment]
I have fixed the issue with URL_SCHEME https://github.com/jguix/Cordova-Flic/pull/3

**Comment by dukhanov** [comment]
Regarding **LSApplicationQueriesSchemes**: 
we should set the param to **flic20** (I have not found explanation regarding that parameter).

Regarding **URL Schemes**:
we should specify that param, because after the user has chosen one Flic button (inside the Flic App), the Flic App will redirect back to the specified callback URL (Our app should catch that redirect).

**Comment by dukhanov** [comment]
Regarding global events:

```
document.addEventListener('flicButtonClick', this.onFlicButtonPressed, false);
document.addEventListener('flicButtonDblClick', this.onFlicButtonPressed, false);
document.addEventListener('flicButtonHold', this.onFlicButtonPressed, false);
```

I don't think that approach is good. I think we don't should use global events, we should subscribe on events from Flic plugin.
some kind of:

```
FlicPlugin.on('flicButtonClick', this.onFlicButtonPressed, false);
FlicPlugin.on('flicButtonDblClick', this.onFlicButtonPressed, false);
FlicPlugin.on('flicButtonHold', this.onFlicButtonPressed, false);
```

I want to add ability to use the Flic events besides the global events.

What do you think about that?

For example https://github.com/chariotsolutions/phonegap-nfc does not use global (document) events

**Pull request merged**